srcalc/src/main/java/gov/va/med/srcalc/web/view/PopulatedDisplayGroup.java

```diff
@@ -61,10 +61,15 @@ public PopulatedDisplayGroup(final List<? extends Variable> variables, final Pat
         fDisplayItems = new ArrayList<>();
         for(final Variable var: variables)
         {
-            // TODO: If the variable has specified keys that have reference information,
-            // create a String that holds that reference information and use it in the view constructors.
             final Visitor visitor = new Visitor();
-            fDisplayItems.add(visitor.getView(var, ""));
+            if(var.getRetriever() != null)
+            {
+                fDisplayItems.add(visitor.getView(var, var.getRetriever().retrieveReferenceInfo(patient, var)));
+            }
+            else
+            {
+                fDisplayItems.add(visitor.getView(var, ""));
```

Comment:
Duplicate code: you call `fDisplayItems.add(visitor.getView())` in both places. Just extract out the different code in the if/then block.